src/Sulu/Bundle/ContentBundle/Resources/public/js/components/roles/texteditor/main.js

```diff
@@ -0,0 +1,199 @@
+
+            this.bindDOMEvents();
+            this.bindCustomEvents();
+        },
+
+        bindDOMEvents: function() {
+            this.sandbox.dom.on('#select-all', 'click', function() {
+                this.sandbox.emit('husky.matrix.set-all');
+            }.bind(this));
+        },
+
+        bindCustomEvents: function() {
+            this.sandbox.on('sulu.header.back', RoleRouter.toList);
+            this.sandbox.on('husky.matrix.changed', this.changeData.bind(this));
+            this.sandbox.on('husky.matrix.changed', function() {
+                this.sandbox.emit('sulu.header.toolbar.item.enable', 'save', false);
+            }.bind(this));
+            this.sandbox.on('sulu.toolbar.save', this.save.bind(this));
+            this.sandbox.on('sulu.toolbar.delete', this.delete.bind(this));
+        },
+
+        render: function() {
+            var values = this.prepareMatrixValues();
+            var data = this.prepareMatrixData();
+
+            this.$el.html(this.templates.skeleton({translations: this.translations, content: this.data.roleModel.toJSON()}));
+
+            this.sandbox.start([
+                {
+                    name: 'matrix@husky',
+                    options: {
+                        el: '#matrix-container',
+                        captions: {
+                            all: this.translations.all,
+                            none: this.translations.none,
+                            type: this.translations.type,
+                            horizontal: this.translations.horizontal,
+                            vertical: values.vertical
+                        },
+                        values: values,
+                        data: data
+                    }
+                }
+            ]);
+        },
+
+        prepareMatrixValues: function() {
+            var vertical = [],
+                horizontal = [];
+
+            for (var key in this.toolbar) {
+                if (!this.toolbar.hasOwnProperty(key)) {
+                    continue;
+                }
+
+                vertical.push(key);
+                horizontal.push(_.map(this.toolbar[key], function(button) {
+                    return {value: button, icon: this.sandbox.ckeditor.getIcon(button), title: button};
+                }.bind(this)));
+            }
+
+            return {vertical: vertical, horizontal: horizontal};
+        },
+
+        prepareMatrixData: function() {
+            var data = [];
+            for (var section in this.toolbar) {
+                data.push(
+                    _.map(this.toolbar[section], function(item) {
+                        return _.contains(this.data.toolbarRights[section], item);
+                    }.bind(this))
+                );
+            }
+
+            return data;
+        },
+
+        changeData: function(data) {
+            if (typeof(data.value) !== 'string') {
+                this.sandbox.dom.each(data.value, function(key, nestedData) {
```

Comment:
`button` or `element` could be misleading as the data comes from the matrix. I chose `nested` as the function flattens the data when it got passed data which consist of other nested data. We could call the function parameter `matrixData` and in the flatten part use `nestedMatrixData` instead of `nestedData`.